dot/types/block_data.go

```diff
@@ -18,196 +18,64 @@ package types
 
 import (
 	"fmt"
-	"io"
 	"math/big"
 
 	"github.com/ChainSafe/gossamer/lib/common"
-	"github.com/ChainSafe/gossamer/lib/common/optional"
-	"github.com/ChainSafe/gossamer/lib/scale"
 )
 
 // BlockData is stored within the BlockDB
 type BlockData struct {
 	Hash          common.Hash
-	Header        *optional.Header
-	Body          *optional.Body
-	Receipt       *optional.Bytes
-	MessageQueue  *optional.Bytes
-	Justification *optional.Bytes
+	Header        *Header
+	Body          *Body
+	Receipt       *[]byte
+	MessageQueue  *[]byte
+	Justification *[]byte
```

Comment:
would be nice to add a comment saying these are optional fields and that's why they are pointers